include/multipass/exceptions/cmd_exceptions.h

```diff
@@ -28,6 +28,12 @@ class ValidationException : public std::runtime_error
     using std::runtime_error::runtime_error;
 };
 
+class ValueException : public std::runtime_error
```

Comment:
Could this be made more specific to the prompter, like `PromptException`? That would clarify the distinction between troubles reading vs interpreting setting values. And it is not conceptually tied to a command: the `PlainPrompter` could in principle be used outside of CLI commands.

Reply:
FWIW I didn't want to introduce another file (we have too many separate exception headers already…). Maybe it could be renamed to `cli_exceptions.h` - CLI does not necessarily mean just our `multipass` command, but generally the Command Line Interface - to which a Prompter, by design, is tied. But then `ValidationException` probably doesn't fit.

And then exception type proliferation… I feel like we could have some "generic" enough exceptions that apply in multiple places. Nothing in a Prompter could throw a Settings-related exception, so do we need a separate `PromptValueException` and `SettingsValueException`?

Don't get me wrong. I'm not saying what you're asking for is wrong. I'm just asking if there's maybe a middle ground that we should be going for?

Reply:
OK, I see what you mean. I guess proliferating exception types is what we pay to avoid error-handling boilerplate in each function call (as you see in Go, eww). With exceptions, you don't need to `if (ret != success) return ret;` in every node of the call tree, but you need a way to distinguish problems arising at different nodes, *if they need different treatment*. Hence the multiple types. It's still a very worthy tradeoff IMO. We may also choose to introduce umbrella headers and revise exception hierarchies at some point.

In general, when programing a reusable unit, one doesn't know how it will be used and how its errors should be handled. One of the main points of exceptions is exactly to delegate error handling up the stack. Because the throwing unit doesn't know if its errors merit specific treatment, the generic approach is to throw a dedicated type. However, since we are the only consumer, we can take the lazy approach of throwing `runtime_error` until we know it needs special treatment. 

I understand you're after the same sort of thing, just a little more refined. I think that makes sense, but it feels to me like there is a naming issue. Perhaps `CLIException` or `TerminalException` would be better names? At face value, `ValueException` could easily arise in situations where the command line isn't at fault (e.g. reading a setting from disk). And yet, it's maybe not the best fit for a failing stream.

> Nothing in a Prompter could throw a Settings-related exception, so do we need a separate `PromptValueException` and `SettingsValueException`?

As with `runtime_error`, you can analyze the call tree and determine that the generic exception is only thrown in one situation today, but that's fragile. When we integrate Prompter and Settings (or whenever we descend into both from a single try block), we may want to distinguish those cases: one of them is not necessarily a cmd line error.

Reply:
OK went for a `PromptException` in a new `cli_exceptions.cpp` in the end.